feat: Document metadata decorators (#203)
Quantinuum/guppy-docs: sphinx/language_guide/custom_metadata.md

---
file_format: mystnb
kernelspec:
  name: python3
---

# Custom Metadata

> Guppy supports adding arbitrary metadata to symbols that will be lowered to the final compilation product (the HUGR package). You can do this via the {py:deco}`guppylang.decorator.metadata` decorator, which you can add below the {py:deco}`guppylang.decorator.guppy` decorator:

Reviewer: ```suggestion
Guppy supports adding arbitrary metadata to symbols that will be lowered to the final compilation product (the HUGR package). This can be done with the {py:deco}`guppylang.decorator.metadata` decorator, which you can add below the {py:deco}`guppylang.decorator.guppy` decorator:
```

Reviewer: Not that important
